core/src/main/scala/stryker4s/run/MutantRunner.scala

```diff
@@ -69,11 +69,41 @@ class MutantRunner(
     val targetDir = config.baseDir / "target"
     for {
       _ <- Resource.eval(Files[IO].createDirectories(targetDir))
-      tmpDir <- Files[IO].tempDirectory(Some(targetDir), "stryker4s-", None)
+      tmpDir <- prepareTmpDir(targetDir)
       _ <- Resource.eval(setupFiles(tmpDir, mutatedFiles.toSeq))
     } yield tmpDir
   }
 
+  private def prepareTmpDir(targetDir: Path): Resource[IO, Path] = {
+    val tmpDirCreated = if (config.staticTmpDir) {
+      val staticTmpDir = targetDir / "stryker4s-tmpDir"
+      Files[IO].createDirectory(staticTmpDir).map(_ => staticTmpDir)
+    } else {
+      Files[IO].createTempDirectory(Some(targetDir), "stryker4s-", None)
+    }
+    Resource.makeCase(tmpDirCreated)(tmpDirFinalizeCase)
+  }
+
+  private val tmpDirFinalizeCase: (Path, Resource.ExitCase) => IO[Unit] = {
```

Comment:
Is there a specific reason you made this a `val` instead of a `def`? I think it would be slightly cleaner as a def since you wouldn't have to have the `case`

Reply:
I think I still need to have the case.

Reply:
Yes, but couldn't it be `def tmpDirFinalizeCase(exitCase....): IO[Unit]`?

Reply:
Switched to a def in any case.

Reply:
> I think I still need to have the case.

I think what I don't need to have is a `match`, but it's also not necessary when defining as a `val`.

Reply:
Like I said, I didn't sleep much last night, but i see now why the `case` is there